OpenMage/magento-lts
# 20.0 paypal paylater messages

### Description (*)
PayPal US has a new feature called Pay Later (or Pay in 4)
This ads in the required messages to inform user that can Pay in 4 instalments, if they want to do so.

ref: https://www.paypal.com/us/for-you/pay-in-4

### Fixed Issues (if relevant)
1. Fixes OpenMage/magento-lts #1435

### Manual testing scenarios (*)

1. In admin setup Paypal Express, must use a US based account (vital)
2. Set Paypal express buttons for Product and Cart to be enabled
2. In admin set Merchant location to US, enable Pay Later messages, set PayPal SDK client id (ref: https://developer.paypal.com/docs/business/pay-later/integrate/)

![image](https://user-images.githubusercontent.com/4994260/106351799-6f71e700-6319-11eb-8f42-cdc17c6011c8.png)

3. In frontend, browse to any product. You will see a message generated that shows the price of the product viewed split PRICE / 4

![image](https://user-images.githubusercontent.com/4994260/106351844-b364ec00-6319-11eb-82e6-f959bc8ea04f.png)

4. Add item to cart (and more items to cart if you so wish)
5. In cart you will get same messaging, based on cart grand total / 4

![image](https://user-images.githubusercontent.com/4994260/106351871-e7401180-6319-11eb-854b-890e62eef501.png)

6. Proceed to checkout. You will get same messaging on payment selection

![image](https://user-images.githubusercontent.com/4994260/106351897-148cbf80-631a-11eb-9409-03876286c22b.png)

7. In paypal wallet, user wil get ability to do the Pay 4 method. It will ONLY work for USA based paypal accounts/merchants

### Questions or comments

First attempt to implement this. Looking for a peer review / suggestions on doing something in code better.
Not sure how to enact unit testing for this.

### Contribution checklist (*)
